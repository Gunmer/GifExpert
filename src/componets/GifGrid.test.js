import {GifGrid} from "./GifGrid";
import {shallow} from "enzyme";
import {useFetchGifs} from "../hooks/useFetchGifs";

jest.mock('../hooks/useFetchGifs')

describe('GifGrid', () => {
  it('should be render', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category="category" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show items when useFetchGifs given', () => {
    const gifs = [{
      id: 'id',
      title: 'Title',
      url: 'http://title.png'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category="category" />);

    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem')).toHaveLength(gifs.length);
    expect(wrapper).toMatchSnapshot();
  });

});
