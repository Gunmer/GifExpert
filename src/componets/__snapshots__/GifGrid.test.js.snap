// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should be render 1`] = `
<Fragment>
  <h3>
    category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid should show items when useFetchGifs given 1`] = `
<Fragment>
  <h3>
    category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="id"
      key="id"
      title="Title"
      url="http://title.png"
    />
  </div>
</Fragment>
`;
